File: Samples for Parallel Programming with the .NET Framework/[C#][VB][C++][F#]-Samples for Parallel Programming with the .NET Framework/C# and C++ and F# and VB/C#,C++,F#,VB/Sudoku/Sudoku_VisualBasic/SolverGenerator/PuzzleDifficulty.vb
'--------------------------------------------------------------------------
' 
'  Copyright (c) Microsoft Corporation.  All rights reserved. 
' 
'  File: PuzzleDifficulty.vb
'
'  Description: Puzzle difficulty enumeration.
' 
'--------------------------------------------------------------------------


Namespace Microsoft.ParallelComputingPlatform.ParallelExtensions.Samples.Sudoku
	''' <summary>The level of difficulty for a puzzle.</summary>
	<Serializable>
	Public Enum PuzzleDifficulty
		''' <summary>An easy puzzle.</summary>
		Easy
		''' <summary>A medium puzzle.</summary>
		Medium
		''' <summary>A hard puzzle.</summary>
		Hard
		''' <summary>A very puzzle.</summary>
		VeryHard
		''' <summary>An unsolvable puzzle, either because it has multiple solutions or no solutions.</summary>
		Invalid
	End Enum
End Namespace

File: Samples for Parallel Programming with the .NET Framework/[C#][VB][C++][F#]-Samples for Parallel Programming with the .NET Framework/C# and C++ and F# and VB/C#,C++,F#,VB/Sudoku/Sudoku_CSharp/SolverGenerator/PuzzleDifficulty.cs
//--------------------------------------------------------------------------
// 
//  Copyright (c) Microsoft Corporation.  All rights reserved. 
// 
//  File: PuzzleDifficulty.cs
//
//  Description: Puzzle difficulty enumeration.
// 
//--------------------------------------------------------------------------

using System;

namespace Microsoft.ParallelComputingPlatform.ParallelExtensions.Samples.Sudoku
{
	/// <summary>The level of difficulty for a puzzle.</summary>
	[Serializable]
	public enum PuzzleDifficulty
	{
		/// <summary>An easy puzzle.</summary>
		Easy,
		/// <summary>A medium puzzle.</summary>
		Medium,
		/// <summary>A hard puzzle.</summary>
		Hard,
        /// <summary>A very puzzle.</summary>
        VeryHard,
		/// <summary>An unsolvable puzzle, either because it has multiple solutions or no solutions.</summary>
		Invalid
	}
}